code/game/objects/items/storage/belt.dm

```diff
@@ -1210,13 +1210,39 @@
 	)
 	holster_slots = list(
 		"1" = list(
-			"icon_x" = 9,
-			"icon_y" = -2))
+			"icon_x" = 10,
+			"icon_y" = -1))
 	item_state_slots = list(
 		WEAR_J_STORE = "m4a3_holster",
 		WEAR_WAIST = "xm51_holster"
 		)
 
+	var/magazines = 0 //Count how many magazines are inside the belt so we can only fit a max of 2.
+
+/obj/item/storage/belt/gun/xm51/attackby(obj/item/item, mob/user)
+	if(istype(item, /obj/item/ammo_magazine/shotgun/light/breaching))
+		var/obj/item/ammo_magazine/shotgun/light/breaching/ammo_box = item
+		dump_ammo_to(ammo_box, user, ammo_box.transfer_handful_amount)
+	else
+		return ..()
+
+/obj/item/storage/belt/gun/xm51/can_be_inserted(obj/item/item, mob/user, stop_messages = FALSE)
+	. = ..()
+	if(magazines == 2 && istype(item, /obj/item/ammo_magazine/rifle/xm51))
+		if(!stop_messages)
+			to_chat(usr, SPAN_WARNING("[src] can't hold any more magazines."))
+		return FALSE
+
+/obj/item/storage/belt/gun/xm51/_item_insertion(obj/item/item, prevent_warning = 0, mob/user)
+	. = ..()
+	if(istype(item, /obj/item/ammo_magazine/rifle/xm51))
+		magazines++
+
+/obj/item/storage/belt/gun/xm51/_item_removal(obj/item/item, prevent_warning = 0, mob/user)
+	. = ..()
+	if(istype(item, /obj/item/ammo_magazine/rifle/xm51))
+		magazines--
```

Comment:
You should not be using the _private methods. Override/implement handle_item_insertion and remove_from_storage instead.